cmd/web3/main.go

```diff
@@ -258,24 +258,35 @@ func parseBigInt(value string) (*big.Int, error) {
 	if value == "" {
 		return nil, nil
 	}
-	i := big.Int{}
-	_, err := fmt.Sscan(value, &i)
-	return &i, err
+	i, ok := new(big.Int).SetString(value, 10)
+	if !ok {
+		return nil, errors.New("failed to parse integer")
+	}
+	return i, nil
```

Comment:
`fmt.Sscan` was accepting e.g. `1234a` as `1234` and ignoring anything trailing.